### PR TITLE
Add AWS IAM Component

### PR DESCRIPTION
**Pull request by zchase**
This PR adds the overview pages and the data file for the AWS IAM package. It wasn't clear to me how the `api-docs` are generated in the pipeline but my assumption was it was generated at the `docs` level during the build. Please let me know if there are additional steps I need to take to get the docs published!

### CODE REVIEW

File: themes/default/data/registry/packages/aws-iam.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+category: Cloud
+component: true
+description: ""
+featured: false
+logo_url: ""
+name: aws-iam
+native: true
```
Comment (stack72) [inline themes/default/data/registry/packages/aws-iam.yaml:7]: This shouldn't be a native - you are missing keywords - see aws-apigateway for the ones you want

**Review by stack72 — COMMENTED** [review]
1 small change needed @zchase

**Comment by susanev** [comment]
@zchase heres a logo for you: https://github.com/pulumi/pulumi-hugo/pull/1582

File: themes/default/data/registry/packages/aws-iam.yaml (hunk 1 above)
Comment (zchase) [reply themes/default/data/registry/packages/aws-iam.yaml]: @stack72 update the `native` value but not sure what else you're referring to. I compared it to the aws-apigateway data file and they look similar now.

**Review by stack72 — APPROVED** [review]

**Pull request merged**